gateway/gateway-requests/src/registration/handshake/state.rs

```diff
@@ -63,40 +119,21 @@ impl<'a, S> State<'a, S> {
     // Eventually the ID_PUBKEY prefix will get removed and recipient will know
     // initializer's identity from another source.
     pub(crate) fn init_message(&self) -> Vec<u8> {
-        self.identity
-            .public_key()
-            .to_bytes()
-            .iter()
-            .cloned()
-            .chain(
-                self.ephemeral_keypair
-                    .public_key()
-                    .to_bytes()
-                    .iter()
-                    .cloned(),
-            )
-            .collect()
+        InitMessage::new(
+            self.identity.public_key(),
+            self.ephemeral_keypair.public_key(),
+            self.coconut_credential.clone().unwrap(),
+        )
+        .to_bytes()
     }
 
     // this will need to be adjusted when REMOTE_ID_PUBKEY is removed
-    pub(crate) fn parse_init_message(
-        mut init_message: Vec<u8>,
-    ) -> Result<(identity::PublicKey, encryption::PublicKey), HandshakeError> {
-        if init_message.len() != identity::PUBLIC_KEY_LENGTH + encryption::PUBLIC_KEY_SIZE {
-            return Err(HandshakeError::MalformedRequest);
+    #[cfg(not(target_arch = "wasm32"))]
```

Comment:
I don't think `#[cfg(not(target_arch = "wasm32"))]` is necessary here?

Reply:
Well, it is currently only used by the gateway server code, so without this you'll get a "function not used" warning.